### PR TITLE
Test current behaviour of cipher suite ordering

### PR DESCRIPTION
**Pull request by yschimke**
First PR to confirm ordering of existing functionality.  Relates to https://github.com/square/okhttp/issues/6390

### CODE REVIEW

File: okhttp/src/test/java/okhttp3/CallHandshakeTest.kt [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+
+  @Test
+  fun testDefaultHandshakeCipherSuiteOrderingTls12Modern() {
+    val client = makeClient(ConnectionSpec.MODERN_TLS, TlsVersion.TLS_1_2)
+
+    val handshake = makeRequest(client)
+
+    assertThat(handshake.cipherSuite).isIn(expectedModernTls12CipherSuites)
+
+    // Probably something like
```
Comment (swankjesse) [inline okhttp/src/test/java/okhttp3/CallHandshakeTest.kt:95]: According to [the doc](https://docs.google.com/spreadsheets/d/1C3FdZSlCBq_-qrVwG1KDIzNIB3Hyg_rKAcgmSzOsHyQ/edit#gid=0) it should be this:

TLS_AES_128_GCM_SHA256
TLS_CHACHA20_POLY1305_SHA256
TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384
TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256
TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256
TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA
TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA
TLS_RSA_WITH_AES_128_GCM_SHA256
TLS_RSA_WITH_AES_256_GCM_SHA384
TLS_RSA_WITH_AES_128_CBC_SHA
TLS_RSA_WITH_AES_256_CBC_SHA
SSL_RSA_WITH_3DES_EDE_CBC_SHA

File: okhttp/src/test/java/okhttp3/CallHandshakeTest.kt [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+
+  @RegisterExtension
+  @JvmField
+  var platform = PlatformRule()
+
+  private lateinit var handshakeEnabledCipherSuites: List<String>
+  private lateinit var defaultEnabledCipherSuites: List<String>
+  private lateinit var defaultSupportedCipherSuites: List<String>
+
+  val expectedModernTls12CipherSuites = listOf(TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384, TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384)
```
Comment (swankjesse) [inline okhttp/src/test/java/okhttp3/CallHandshakeTest.kt:49]: why is this list so short?

File: okhttp/src/test/java/okhttp3/CallHandshakeTest.kt [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+    // TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
+    // TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
+    // TLS_RSA_WITH_AES_256_GCM_SHA384
+    // TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
+    // TLS_RSA_WITH_AES_128_GCM_SHA256
+    // TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA
+    // TLS_RSA_WITH_AES_256_CBC_SHA
+    // TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA
+    // TLS_RSA_WITH_AES_128_CBC_SHA
+    assertThat(handshakeEnabledCipherSuites).containsExactlyElementsOf(expectedConnectionCipherSuites(client))
```
Comment (swankjesse) [inline okhttp/src/test/java/okhttp3/CallHandshakeTest.kt:136]: this isn’t asserting the order. Is that unexpected?

**Review by swankjesse — COMMENTED** [review]
Good start but this should call out that it’s asserting what we do, not necessarily what we want.

**Review by swankjesse — APPROVED** [review]

**Comment by yschimke** [comment]
> 
> 
> Good start but this should call out that it’s asserting what we do, not necessarily what we want.

Consider this a toehold test.

I think two follow up streams

1) Refine our current and general expectations -> TODO in follow ups
2) Fix the bug -> https://github.com/yschimke/okhttp/pull/6/files

File: okhttp/src/test/java/okhttp3/CallHandshakeTest.kt (hunk 3 above)
Comment (yschimke) [reply okhttp/src/test/java/okhttp3/CallHandshakeTest.kt:136]: I'm going to land with it more relaxed and passing then we can define exactly what our current and future expectations are.

File: okhttp/src/test/java/okhttp3/CallHandshakeTest.kt (hunk 2 above)
Comment (yschimke) [reply okhttp/src/test/java/okhttp3/CallHandshakeTest.kt:49]: Given a fixed set of platforms we test against in CI, there are only a certain number of likely results for the first cipher.  But at least if we curate any valid high strength choices here, it's a smoke test if it ever breaks and starts using a weaker one.

File: okhttp/src/test/java/okhttp3/CallHandshakeTest.kt (hunk 3 above)
Comment (yschimke) [reply okhttp/src/test/java/okhttp3/CallHandshakeTest.kt:136]: It is asserting the order according to javadoc

```
   * Same as {@link #containsExactly(Object...)} but handle the {@link Iterable} to array conversion : verifies that
   * actual contains exactly the elements of the given iterable and nothing else <b>in the same order</b>.
```

**Pull request merged**